File: src/dotnet/Ansight/Pairing/Models/PairingControlEnvelope.cs
using System.Text.Json.Nodes;

namespace Ansight.Pairing.Models;

/// <summary>
/// Request/response envelope used for correlated control messages on the live pairing WebSocket session.
/// </summary>
public sealed class PairingControlEnvelope
{
    public const string RequestType = "CONTROL_REQ";
    public const string ResponseType = "CONTROL_RESP";

    /// <summary>
    /// Message type identifier.
    /// </summary>
    public required string Type { get; set; }

    /// <summary>
    /// Request identifier for outbound requests.
    /// </summary>
    public string? Id { get; set; }

    /// <summary>
    /// Request identifier that this response answers.
    /// </summary>
    public string? ReplyTo { get; set; }

    /// <summary>
    /// Logical action being requested or answered.
    /// </summary>
    public required string Action { get; set; }

    /// <summary>
    /// Optional structured payload for the action.
    /// </summary>
    public JsonObject? Payload { get; set; }

    /// <summary>
    /// Indicates whether the request succeeded.
    /// </summary>
    public bool Success { get; set; } = true;

    /// <summary>
    /// Human-readable status message.
    /// </summary>
    public string? Message { get; set; }
}

File: src/ios/Sources/AnsightCore/PairingControlEnvelope.swift
import Foundation
import Network

public struct PairingControlEnvelope: Sendable, Codable, Equatable {
    public static let requestType = "CONTROL_REQ"
    public static let responseType = "CONTROL_RESP"

    public let type: String
    public let id: String?
    public let replyTo: String?
    public let action: String
    public let payload: JSONValue?
    public let success: Bool
    public let message: String?

    public init(
        type: String,
        id: String? = nil,
        replyTo: String? = nil,
        action: String,
        payload: JSONValue? = nil,
        success: Bool = true,
        message: String? = nil
    ) {
        self.type = type
        self.id = id
        self.replyTo = replyTo
        self.action = action
        self.payload = payload
        self.success = success
        self.message = message
    }
}
